src/components/GLStyleService.js

```diff
@@ -0,0 +1,122 @@
+goog.provide('ga_gl_style_service');
+(function() {
+  var module = angular.module('ga_gl_style_service', []);
+
+  module.provider('gaGLStyle', function() {
+    this.$get = function($http, $q, $window) {
+      var GLStyle = function() {
+        this.styleJSONCache = null;
```

Comment:
the name is a bit confusing can we just name it  'glStyle'   so we knew it's the complete glStyle  in a js object